test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -12,234 +12,235 @@
-		podData = append(podData, output)
-	}
-	return nil
-}
-
 // formAnnotationStringOfPod forms the annotation string, used in the generation of each Pod YAML file.
-func (data *TestData) formAnnotationStringOfPod(pod int) string {
+func (data *testData) formAnnotationStringOfPod(pod *testPodInfo) string {
 	var annotationString = ""
-	for xPodVN := 0; xPodVN < podData[pod].countOfVirtualNetworksPerPod; xPodVN++ {
-		var podNetworkSpec = "{\"name\": \"" + podData[pod].nameOfVirtualNetworkPerPod[xPodVN] + "\" ,\"interface\": \"" + podData[pod].nameOfInterfacePerPod[xPodVN] + "\" , \"type\": \"" + interfaceType + "\"}"
+	for i, n := range pod.interfaceNetworks {
+		podNetworkSpec := fmt.Sprintf("{\"name\": \"%s\", \"namespace\": \"%s\", \"interface\": \"%s\"}",
+			n, attachDefNamespace, i)
 		if annotationString == "" {
 			annotationString = "[" + podNetworkSpec
 		} else {
-			annotationString = annotationString + "," + podNetworkSpec
+			annotationString = annotationString + ", " + podNetworkSpec
 		}
 	}
 	annotationString = annotationString + "]"
 	return annotationString
 }
 
 // createPodOnNode creates the Pod for the specific annotations as per the parsed Pod information using the NewPodBuilder API
-func (data *TestData) createPodOnNode(t *testing.T, ns string, nodeName string) error {
+func (data *testData) createPods(t *testing.T, ns string) error {
 	var err error
-	for xPod := 0; xPod < totalNumberOfPods; xPod++ {
-		err := data.createPodForSecondaryNetwork(ns, nodeName, xPod, testPodName, resNum)
+	for _, pod := range data.pods {
+		err := data.createPodForSecondaryNetwork(ns, pod)
 		if err != nil {
-			return fmt.Errorf("Error in creating pods.., err: %v", err)
+			return fmt.Errorf("error in creating pods.., err: %v", err)
 		}
 	}
 	return err
 }
 
-// getSecondaryInterface shows up the secondary interfaces created for the specific Pod and extracts the IP address for the same.
-func (data *TestData) getSecondaryInterface(targetPod int, targetInterface int) (string, error) {
-	cmd := []string{"/bin/sh", "-c", fmt.Sprintf("ip addr show %s | grep \"inet\" | awk '{print $2}' | cut -d/ -f1", podData[targetPod].nameOfInterfacePerPod[targetInterface])}
-	stdout, _, err := data.e2eTestData.RunCommandFromPod(nameSpace, podData[targetPod].nameOfPods, ctrName, cmd)
+// The Wrapper function createPodForSecondaryNetwork creates the Pod adding the annotation, arguments, commands, Node, container name,
+// resource requests and limits as arguments with the NewPodBuilder API
+func (data *testData) createPodForSecondaryNetwork(ns string, pod *testPodInfo) error {
+	podBuilder := antreae2e.NewPodBuilder(pod.podName, ns, antreae2e.BusyboxImage).
+		OnNode(pod.nodeName).WithContainerName(containerName).
+		WithCommand([]string{"sleep", "infinity"}).
```

Comment:
If you switch to antrea/toolbox, you will not need this as it is the default command

Reply:
Ok. Changed.